docs/design/lexical_conventions/comments.md

```diff
@@ -0,0 +1,156 @@
+# Comments
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Comments](#comments)
+    -   [Table of contents](#table-of-contents)
+    -   [Overview](#overview)
+    -   [Details](#details)
+    -   [Alternatives considered](#alternatives-considered)
+        -   [Intra-line comments](#intra-line-comments)
+        -   [Block comments](#block-comments)
+        -   [Documentation comments](#documentation-comments)
+        -   [Code folding comments](#code-folding-comments)
+    -   [References](#references)
+
+<!-- tocstop -->
+
+## Overview
+
+A comment is a lexical element beginning with the characters `//` and running to
+the end of the line. We have no mechanism for physical line continuation, so a
+trailing `\` does not extend a comment to subsequent lines.
+
+## Details
+
+In the comments after the `//` a whitespace character is required to make the
+comment valid. Newline is a whitespace character, so a line containing only `//`
+is a valid comment. The end of the file also constitutes whitespace.
+
+All comments are removed prior to formation of tokens.
+
+Example:
+
+```
+// This is a comment and is ignored. \
+This is not a comment.
+
+var Int: x; // error, trailing comments not allowed
+```
+
+Currently no support for block comments is provided. Commenting out larger
+regions of human-readable text or code is accomplished by commenting out every
+line in the region.
+
+## Alternatives considered
```

Comment:
In the design, we're providing alternatives as links back to the original proposal, rather than copied in. For an example, see: https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/lexical_conventions/string_literals.md?plain=1#L320

Reply:
I added the links and they are working but in pre-commit, this was been shown:
![image](https://user-images.githubusercontent.com/72661784/224465748-e26b2d36-738a-453b-9340-88e70fde44d5.png)

Reply:
@chandlerc Actually I fixed all those existing issues relating to `pre-commit`, but because of this issue I am not able to push it! I have made a similar mistake in the other PR too, thought I would fix this first and then would change the other.

It's saying the links don't exist, but it is there! what am I doing wrong here? Its also states that one source file doesn't exist, I didn't touch that still it's saying that! thats why I didn't push it! What am I doing wrong here?